### ISSUE TITLE
Error during data composer stage

### ISSUE THREAD

**Issue by dhombios** [Nuitka/Nuitka#935]
Trying to use Nuitka for generating an executable from a Tkinter python app the process crashes after stating the data composer tool. However, it seems that a similar issue was fixed in Nuitka 6.0.6: https://github.com/Nuitka/Nuitka/issues/477

Platform: Windows (Anaconda)
Python: 3.7.9 (x32)
Nuitka: 0.6.10.5

CMD output:
```
nuitka --follow-imports --standalone --plugin-enable=pylint-warnings --plugin-enable=tk-inter --plugin-enable=numpy --plugin-enable=qt-plugins main.py
Nuitka:INFO: Starting Python compilation.
Nuitka-Plugins:INFO: tk-inter: Injecting pre-module load code for module 'tkinter':
Nuitka-Plugins:INFO: tk-inter:     Need to make sure we set environment variables for TCL.
Nuitka-Plugins:INFO: numpy: Injecting pre-module load code for module 'matplotlib':
Nuitka-Plugins:INFO: numpy:     Setting 'MATPLOTLIBDATA' environment variable for matplotlib to find package data.
Nuitka-Plugins:INFO: qt-plugins: Injecting post-module load code for module 'PyQt5.QtCore':
Nuitka-Plugins:INFO: qt-plugins:     Setting Qt library path to distribution folder. Need to avoid loading target
Nuitka-Plugins:INFO: qt-plugins:     system Qt plug-ins, which may be from another Qt version.
Nuitka-Plugins:INFO: qt-plugins: Injecting pre-module load code for module 'PyQt5':
Nuitka-Plugins:INFO: qt-plugins:     Adding binary folder to runtime 'PATH' environment variable for proper loading.
Nuitka:INFO: Completed Python level compilation and optimization.
Nuitka:INFO: Generating C source code for backend compiler.
Nuitka:INFO: Running data composer tool for optimal constant value handling.
Traceback (most recent call last):
  File "D:\Dhom\Anaconda3\envs\uC_Testbench\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "D:\Dhom\Anaconda3\envs\uC_Testbench\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "D:\Dhom\Anaconda3\envs\uC_Testbench\lib\site-packages\nuitka\tools\data_composer\__main__.py", line 37, in <module>
    import re
  File "D:\Dhom\Anaconda3\envs\uC_Testbench\lib\re.py", line 145, in <module>
    class RegexFlag(enum.IntFlag):
AttributeError: module 'enum' has no attribute 'IntFlag'
Traceback (most recent call last):
  File "D:\Dhom\Anaconda3\envs\uC_Testbench\lib\site-packages\nuitka\__main__.py", line 196, in <module>
    main()
  File "D:\Dhom\Anaconda3\envs\uC_Testbench\lib\site-packages\nuitka\__main__.py", line 189, in main
    MainControl.main()
  File "D:\Dhom\Anaconda3\envs\uC_Testbench\lib\site-packages\nuitka\MainControl.py", line 701, in main
    result, options = compileTree(main_module=main_module)
  File "D:\Dhom\Anaconda3\envs\uC_Testbench\lib\site-packages\nuitka\MainControl.py", line 629, in compileTree
    runDataComposer(source_dir)
  File "D:\Dhom\Anaconda3\envs\uC_Testbench\lib\site-packages\nuitka\build\DataComposerInterface.py", line 38, in runDataComposer
    shell=False,
  File "D:\Dhom\Anaconda3\envs\uC_Testbench\lib\subprocess.py", line 363, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['D:\\Dhom\\Anaconda3\\envs\\uC_Testbench\\python.exe', 'D:\\Dhom\\Anaconda3\\envs\\uC_Testbench\\lib\\site-packages\\nuitka\\tools\\data_composer', 'main.build', 'main.build\\__constants.bin']' returned non-zero exit status 1.
```

I've also tried updating to python 3.8.5, but the same error occurs:
```
nuitka --follow-imports --standalone --plugin-enable=pylint-warnings --plugin-enable=tk-inter --plugin-enable=numpy --plugin-enable=qt-plugins main.py
Nuitka:INFO: Starting Python compilation.
Nuitka-Plugins:INFO: tk-inter: Injecting pre-module load code for module 'tkinter':
Nuitka-Plugins:INFO: tk-inter:     Need to make sure we set environment variables for TCL.
Nuitka-Plugins:INFO: numpy: Injecting pre-module load code for module 'matplotlib':
Nuitka-Plugins:INFO: numpy:     Setting 'MATPLOTLIBDATA' environment variable for matplotlib to find package data.
Nuitka-Plugins:INFO: qt-plugins: Injecting post-module load code for module 'PyQt5.QtCore':
Nuitka-Plugins:INFO: qt-plugins:     Setting Qt library path to distribution folder. Need to avoid loading target
Nuitka-Plugins:INFO: qt-plugins:     system Qt plug-ins, which may be from another Qt version.
Nuitka-Plugins:INFO: qt-plugins: Injecting pre-module load code for module 'PyQt5':
Nuitka-Plugins:INFO: qt-plugins:     Adding binary folder to runtime 'PATH' environment variable for proper loading.
Nuitka:INFO: Completed Python level compilation and optimization.
Nuitka:INFO: Generating C source code for backend compiler.
Nuitka:INFO: Running data composer tool for optimal constant value handling.
Traceback (most recent call last):
  File "D:\Dhom\Anaconda3\envs\py38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "D:\Dhom\Anaconda3\envs\py38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "D:\Dhom\Anaconda3\envs\py38\lib\site-packages\nuitka\tools\data_composer\__main__.py", line 37, in <module>
    import re
  File "D:\Dhom\Anaconda3\envs\py38\lib\re.py", line 145, in <module>
    class RegexFlag(enum.IntFlag):
AttributeError: module 'enum' has no attribute 'IntFlag'
Traceback (most recent call last):
  File "D:\Dhom\Anaconda3\envs\py38\lib\site-packages\nuitka\__main__.py", line 196, in <module>
    main()
  File "D:\Dhom\Anaconda3\envs\py38\lib\site-packages\nuitka\__main__.py", line 189, in main
    MainControl.main()
  File "D:\Dhom\Anaconda3\envs\py38\lib\site-packages\nuitka\MainControl.py", line 701, in main
    result, options = compileTree(main_module=main_module)
  File "D:\Dhom\Anaconda3\envs\py38\lib\site-packages\nuitka\MainControl.py", line 629, in compileTree
    runDataComposer(source_dir)
  File "D:\Dhom\Anaconda3\envs\py38\lib\site-packages\nuitka\build\DataComposerInterface.py", line 31, in runDataComposer
    subprocess.check_call(
  File "D:\Dhom\Anaconda3\envs\py38\lib\subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['D:\\Dhom\\Anaconda3\\envs\\py38\\python.exe', 'D:\\Dhom\\Anaconda3\\envs\\py38\\lib\\site-packages\\nuitka\\tools\\data_composer', 'main.build', 'main.build\\__constants.bin']' returned non-zero exit status 1.
```

Edit: changed Nutika version from 6.10.5 to 0.6.10.5

**Comment by kayhayen** [comment]
This is by having a enum module in PYTHONPATH that is not the one in the standard library. I can get that by having an empty `enum.py` and that triggers up either scons or Nuitka, but I don't quite get how it's only happening then. Importing the `re` module is something they all do, and getting the wrong kind of `enum` by an absolute import is of course not good. I wonder if there is some wrong doing in Nuitka that makes `PYTHONPATH` end up in front of things.

**Comment by kayhayen** [comment]
So, Nuitka runners now remove themselves from the `sys.path` after loading the `nuitka` package from the desired place, and then there cannot be an impact anymore. Also removed the `bin` directory where they are installed, as it's not needed or useful too.

Applying a forwarding of the nuitka package home directory to data composer allows it to also only temporarily add the path, this should help.

I believe the reason you are not getting it before data composer is by following the recommendation and using `-m nuitka` execution, which avoids the runners.

On factory, there are now these changes, and neither scons, nor data composer, nor nuitka seem to be harmed by an empty `enum.py` anymore.

Generally this should make it only cleaner. I am currently making invasive changes on factory, but this will be in the next release, which I hope to get out "real soon now" /TM/.

**Comment by dhombios** [comment]
Thanks, that was pretty quick. I’ll try using the factory version and I’ll look forward to the next release as the possibility of compiling Python is pretty interesting

**Comment by kayhayen** [comment]
This is in the current pre-release.

**Comment by kayhayen** [comment]
Released as part of 0.6.11
